core/src/main/scala/cromwell/core/retry/Retry.scala

```diff
@@ -61,21 +62,22 @@ object Retry {
    * backoff strategy until a designated number of retries is reached.
    *
    * @param f A function Unit => Future which will be executed once per cycle.
-   * @param maxRetries An optional number of times to retry the future. If this is None, there is no limit.
+   * @param maxRetries Number of times to retry the future
    * @param backoff An exponential backoff strategy to use for each retry strategy.
    * @tparam A The return type of the Future
    * @return The final completed Future[A]
    */
   def withRetryForTransactionRollback[A](f: () => Future[A],
-                                         maxRetries: Option[Int] = Option(5),
+                                         maxRetries: Int = 5,
                                          backoff: Backoff = SimpleExponentialBackoff(5 seconds, 10 seconds, 1.1D))
                                         (implicit actorSystem: ActorSystem, ec: ExecutionContext): Future[A] = {
     val delay = backoff.backoffMillis.millis
 
     f() recoverWith {
       case throwable if throwable.isInstanceOf[SQLTransactionRollbackException] =>
-        val retriesLeft = maxRetries map { _ - 1 }
-        if (retriesLeft.forall(_ > 0)) {
+        val retriesLeft = maxRetries - 1
+        if (retriesLeft > 0) {
+          logger.info(s"Received 'SQLTransactionRollbackException'. Retries left $retriesLeft. Will retry now...")
```

Comment:
Is it worth moving the `"Received 'SQLTransactionRollbackException'` outside the `if` so that we get the log message even if we ran out of retries?

Reply:
If we run out of retries, the calling method would handle the exception and log it. Just like it was doing before we introduced retries.

Reply:
👍